src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -236,6 +258,41 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Hide or show the bell indicator in the tab header
+    // Arguments:
+    // - show: if true, we show the indicator; if false, we hide the indicator
+    winrt::fire_and_forget TerminalTab::ShowBellIndicator(const bool show)
+    {
+        auto weakThis{ get_weak() };
+
+        co_await winrt::resume_foreground(TabViewItem().Dispatcher());
+
+        if (auto tab{ weakThis.get() })
+        {
+            tab->_headerControl.BellIndicator(show);
+        }
+    }
+
+    // Method Description:
+    // - Activates the timer for the bell indicator in the tab
+    // - Called if a bell raised when the tab already has focus
+    winrt::fire_and_forget TerminalTab::ActivateBellIndicatorTimer()
+    {
+        auto weakThis{ get_weak() };
+
+        co_await winrt::resume_foreground(TabViewItem().Dispatcher());
+
+        if (auto tab{ weakThis.get() })
+        {
+            DispatcherTimer bellIndicatorTimer;
+            bellIndicatorTimer.Interval(std::chrono::milliseconds(2000));
+            bellIndicatorTimer.Tick({ get_weak(), &TerminalTab::_BellIndicatorTimerTick });
+            bellIndicatorTimer.Start();
+            tab->_bellIndicatorTimer.emplace(std::move(bellIndicatorTimer));

```

Comment:
can an application DOS us by spamming bell? it will make us start a huge number of timers, and keep overwriting the optional.

Reply:
(we'll waste memory for up to 2 seconds at which point they'll start to fire)

Reply:
Oh yes good catch! We now only create a new timer if the optional doesn't have a value